### DISCUSSION TITLE
[OP Stack Deployment] How to deploy faucet on L2

### DISCUSSION THREAD

**Discussion by avenbreaks** [ethereum-optimism/developers#26] [Deploying the OP Stack] [answered]
### Issue Description

How to deploy faucet on L2 ? Im already bridging and send the tokens using address L1StandardBridgeProxy.json and received on L2

**Comment by opfocus** [comment]
Do you mean erc20 token or eth

**Reply by avenbreaks** [reply]
yes i mean eth on my L2 chain,

**Comment by opfocus** [comment]
I think ETH on the OP-stack chain can only be bridged from L1 to L2. So, the required ETH will need to be bridged to L2

**Reply by avenbreaks** [reply]
how i create the faucet ?

**Comment by sbvegan** [comment, marked as answer]
I believe most faucets are server side applications that use Ethereum libraries to distribute small amounts of ether. Here’s one open source faucet repository I found: https://github.com/komputing/FaucETH. If you search online, I’m sure there are guides and other open source projects you can reference.

**Comment by imtipi** [comment]
i recommend https://github.com/pk910/PoWFaucet
it has some good modules

**Reply by avenbreaks** [reply]
Thank you for the module sir, so I have to build the pow faucet on my network is that right sir?

**Reply by sbvegan** [reply]
Yeah thats right, you'd have to maintain the faucet on a web server and configure it for your network.

**Reply by imtipi** [reply]
really depends on you,if you running a op stack chain that base on mainnet,just provide the bridge and let user bridge the funds.only use faucet when you are running on testnet because you don't need to pay real money

**Reply by sbvegan** [reply]
Yeah thats a good alternative! With this method, you just need to maintain a bridge UI and you can lean on existing testnet faucets.

**Comment by sbvegan** [comment]
Hey @avenbreaks, just following up here. Can you please mark the response that answered your question?

**Reply by sbvegan** [reply]
I'm going to mark this as answered. Please if you think this doesn't answer your question, we can re-open it. If you have a different question, please open a new question.
